src/lib/features/feature-search/feature-search-store.ts

```diff
@@ -295,34 +295,43 @@ class FeatureSearchStore implements IFeatureSearchStore {
         };
     }
 
-    getAggregatedSearchData(rows): IFeatureOverview {
+    getAggregatedSearchData(rows) {
         return rows.reduce((acc, row) => {
-            if (acc[row.feature_name] !== undefined) {
-                const environmentExists = acc[
-                    row.feature_name
-                ].environments.some(
-                    (existingEnvironment) =>
-                        existingEnvironment.name === row.environment,
-                );
-                if (!environmentExists) {
-                    acc[row.feature_name].environments.push(
+            const existingEntry = acc.find(
+                (entry) => entry.name === row.feature_name,
+            );
+
+            if (existingEntry) {
+                if (
+                    !existingEntry.environments.some(
+                        (e) => e.name === row.environment,
+                    )
+                ) {
+                    existingEntry.environments.push(
                         FeatureSearchStore.getEnvironment(row),
                     );
                 }
 
-                const segmentExists = acc[row.feature_name].segments.includes(
-                    row.segment_name,
-                );
+                if (
+                    row.segment_name &&
+                    !existingEntry.segments.includes(row.segment_name)
+                ) {
+                    existingEntry.segments.push(row.segment_name);
+                }
 
-                if (row.segment_name && !segmentExists) {
-                    acc[row.feature_name].segments.push(row.segment_name);
+                if (this.isNewTag(existingEntry, row)) {
+                    this.addTag(existingEntry, row);
                 }
 
-                if (this.isNewTag(acc[row.feature_name], row)) {
-                    this.addTag(acc[row.feature_name], row);
+                if (
+                    !existingEntry.lastSeenAt ||
+                    new Date(row.env_last_seen_at) >
+                        new Date(existingEntry.last_seen_at)
+                ) {
+                    existingEntry.lastSeenAt = row.env_last_seen_at;
                 }
             } else {
-                acc[row.feature_name] = {
```

Comment:
This was the issue. If the key is number, it will be put as first element which breaks the ordering. This means, we need to use actual array instead.